yt_dlp/extractor/openrec.py

```diff
@@ -0,0 +1,127 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import ExtractorError, try_get, unified_strdate
+
+
+class OpenRecBaseIE(InfoExtractor):
+    pass
+
+
+class OpenRecIE(OpenRecBaseIE):
+    IE_NAME = 'openrec'
+    _VALID_URL = r'https?://(?:www\.)?openrec\.tv/live/(?P<id>[^/]+)'
+    _TESTS = [{
+        'url': 'https://www.openrec.tv/live/2p8v31qe4zy',
+        'only_matching': True,
+    }, {
+        'url': 'https://www.openrec.tv/live/wez93eqvjzl',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage('https://www.openrec.tv/live/%s' % video_id, video_id)
+
+        window_stores = self._parse_json(
+            self._search_regex(r'(?m)window\.pageStore\s*=\s*(\{.+?\});$', webpage, 'window.pageStore'), video_id)
+        movie_store = try_get(
+            window_stores,
+            (lambda x: x['v8']['state']['movie'], lambda x: x['v8']['movie']), None)
+        if not movie_store:
+            raise ExtractorError('Failed to extract live info')
+
+        title = movie_store.get('title')
+        description = movie_store.get('introduction')
+        thumbnail = movie_store.get('thumbnailUrl')
+
+        channel_user = movie_store.get('channel', {}).get('user')
+        if channel_user:
+            uploader = channel_user['name']
+            uploader_id = channel_user['id']
+        else:
+            self.report_warning('Failed to extract uploader information')
+            uploader = channel_user.get('name')
+            uploader_id = channel_user.get('id')
```

Comment:
Why so complicated?
Isnt this the same as L106-108?

Reply:
I didn't notice while refactoring the code, will fix